internal/fs/vss_windows.go

```diff
@@ -741,7 +741,9 @@ func NewVssSnapshot(
 		}
 		return VssSnapshot{}, err
 	}
-	defer oleIUnknown.Release()
+	if oleIUnknown != nil {
```

Comment:
Please move this if block before the `if err != nil` block and remove the then redundant call to `Release` inside the latter if block.

Reply:
Fixed in https://github.com/restic/restic/pull/3045/commits/767a686f4032f25d06f81f165bda3f6fee80329c

Reply:
Now that initializeVssCOMInterface check for nil, this function no longer needs to.

```
oleIUnknown, result, err := initializeVssCOMInterface()
if err != nil {
	// handle error
}
defer oleIUnknown.Release()
```

HasSufficientPrivilegesForVSS can do the same without the defer.